### PR TITLE
Refactor smells module.

### PR DESCRIPTION
**Pull request by troessner**

### CODE REVIEW

File: lib/reek/smells.rb [hunk 1]
```diff
@@ -1,28 +1,8 @@
...
-require 'reek/smells/uncommunicative_module_name'
-require 'reek/smells/uncommunicative_parameter_name'
-require 'reek/smells/uncommunicative_variable_name'
-require 'reek/smells/unused_parameters'
-require 'reek/smells/utility_function'
-# SMELL: Duplication -- all these should be found automagically
+smell_glob = Pathname.new('..').
+             join('smells').
+             expand_path(__FILE__).
+             join('*.rb')
```
Comment (mvz) [inline lib/reek/smells.rb]: It seems odd to start with `'..'`...

Comment (mvz) [inline lib/reek/smells.rb]: How about this:

```
smell_glob = Pathname.new(__FILE__).dirname.realpath.join('smells', '*.rb')
```

**Comment by mvz** [comment]
Even nicer: https://github.com/jarmo/require_all

**Comment by mvz** [comment]
If I read the docs correctly, that would just be:

```
require_rel 'smells'
```

for our case.

**Comment by troessner** [comment]
>> Even nicer: https://github.com/jarmo/require_all

I know require_all and I think its awesome -  I just didn't want add more dependencies to reek. 
But maybe we shouldn't be so strict with that if the gem itself is small and self-contained.

Unless anybody objects I'll integrate it.

**Comment by mvz** [comment]
:+1:

**Comment by troessner** [comment]
Ok, this looks a lot more awesome with require_all. Please re-review.:)

**Pull request merged**